### DISCUSSION TITLE
请教下为什么admin service单独设计出一个模块

### DISCUSSION THREAD

**Discussion by shuohao** [apolloconfig/apollo#3788] [Q&A]
看了下设计，portal是用户界面，用户界面发布修改配置调用 admin service，admin service插入db数据库 配置变更记录，config service去扫描库，然后基于长连接通知客户端；感觉这个admin service完全可以和portal合到一起，这个单独设计出来的目的是基于什么考虑的呢？admin service只服务于界面端portal，请求量应该很少，感觉没必要通过服务发现获取admin  service调用，直接取消掉这个模块，通过portal去操作db

**Comment by Anilople** [comment]
多环境的时候，例如有SIT，UAT等10个环境

如果只有1个portal，那么这个portal需要知道所有环境的configdb，也就是

1个portal需要连接1 portaldb + 10 个 configdb

但是抽象出admin service之后，portal可以不用管背后的10个configdb

只需要知道apollo.meta就好了

**Reply by shuohao** [reply]
但是多环境的话portal也是可以对应多个啊

**Reply by pengweiqhca** [reply]
> 但是多环境的话portal也是可以对应多个啊

你想安装多个portal且需要在不同的portal之间切换？

**Reply by shuohao** [reply]
比如我有2个环境 一个测试一个线上，那每个环境可以部署2个或3个portal通过域名配置ng做集群高可用，毕竟线上和测试数据网段等都是隔离，和1个portal 对应多个admin service没啥区别吧
